airflow/models/dag.py

```diff
@@ -770,6 +770,16 @@ def get_is_paused(self, session=NEW_SESSION) -> None:
         """Return a boolean indicating whether this DAG is paused."""
         return session.scalar(select(DagModel.is_paused).where(DagModel.dag_id == self.dag_id))
 
+    @provide_session
+    def get_bundle_name(self, session=NEW_SESSION) -> str:
```

Comment:
Is it possible to be None?

Reply:
It could if the DAG is not synced into the db. This one and `get_latest_bundle_version` should have the same return type (either both `str` or both `str | None`), and if it’s to return `str`, it should use `scalars(...).one()` instead to raise an exception when there’s no matching entry.

Reply:
Possible right now - yes. Possible next week, no :)

For now bundle is nullable in the db, but that's on my list and very near the to

The PR these were split out of is already merged, so I actually need to close this PR. But I'll open one to do a little cleanup on these methods. Thanks!

Reply:
Here is that cleanup PR, #45672. I've left it returning None and will tackle the retyping it later this week.